docs/sources/flow/getting-started/migrating-from-promtail.md

```diff
@@ -0,0 +1,210 @@
+[loki.source.file]: {{< relref "../reference/components/loki.source.file.md" >}}
+[loki.write]: {{< relref "../reference/components/loki.write.md" >}}
+
+## Before you begin
+
+* You must have an existing Promtail configuration.
+* You must be familiar with the concept of [Components][] in Grafana Agent Flow
+  Mode.
+
+[Components]: {{< relref "../concepts/components.md" >}}
+[convert]: {{< relref "../reference/cli/convert.md" >}}
+[run]: {{< relref "../reference/cli/run.md" >}}
+[Start the agent]: {{< relref "../setup/start-agent.md" >}}
+[Flow Debugging]: {{< relref "../monitoring/debugging.md" >}}
+[debugging]: #debugging
+
+## Convert a Promtail configuration
+
+To fully migrate from [Promtail] to Grafana Agent in Flow Mode, you must convert
+your Promtail configuration into a Grafana Agent Flow Mode configuration. This
+conversion will enable you to take full advantage of the many additional
+features available in Grafana Agent Flow Mode.
+
+> In this task, we will use the [convert][] CLI command to output a flow
+> configuration from a Promtail configuration.
+
+1. Open a terminal window and run the following command:
+
+    ```bash
+    AGENT_MODE=flow; grafana-agent convert --source-format=promtail --output=OUTPUT_CONFIG_PATH INPUT_CONFIG_PATH
+    ```
+
+   Replace the following:
+    * `INPUT_CONFIG_PATH`: The full path to the Promtail configuration.
+    * `OUTPUT_CONFIG_PATH`: The full path to output the flow configuration.
+
+1. [Start the agent][] in Flow Mode using the new flow configuration
+   from `OUTPUT_CONFIG_PATH`:
+
+### Debugging
+
+1. If the convert command cannot convert a Promtail configuration, diagnostic
+   information is sent to `stderr`. You can bypass any non-critical issues and
+   output the flow configuration using a best-effort conversion by including
+   the `--bypass-errors` flag.
+
+   {{% admonition type="caution" %}}If you bypass the errors, the behavior of the converted configuration may not match the original Promtail configuration. Make sure you fully test the converted configuration before using it in a production environment.{{% /admonition %}}
+
+    ```bash
+    AGENT_MODE=flow; grafana-agent convert --source-format=promtail --bypass-errors --output=OUTPUT_CONFIG_PATH INPUT_CONFIG_PATH
+    ```
+
+1. You can also output a diagnostic report by including the `--report` flag.
+
+    ```bash
+    AGENT_MODE=flow; grafana-agent convert --source-format=promtail --report=OUTPUT_REPORT_PATH --output=OUTPUT_CONFIG_PATH INPUT_CONFIG_PATH
+    ```
+
+    * Replace `OUTPUT_REPORT_PATH` with the output path for the report.
+
+   Using the [example](#example) Promtail configuration below, the diagnostic
+   report provides the following information:
+
+    ```plaintext
+    (Warning) If you have a tracing set up for Promtail, it cannot be migrated to Flow Mode automatically - please refer to documentation on how to configure tracing in Flow Mode
```

Comment:
```suggestion
    (Warning) If you have a tracing set up for Promtail, it cannot be migrated to Flow Mode automatically. Refer to the documentation on how to configure tracing in Flow Mode.
```

I've also suggested the same change in the go files.

1. Generally we should avoid using `please` in technical documentation. It's not wrong, but it is also not necessary. Keep things simple and concise. (also, using `please` implies we are asking the user for a favour vs instructing them on the task).
1. Does the user know where this reference documentation is located?

Reply:
Re: 2. They should ideally be able to find this with ease in our docs. Hard-coding a link in the binary may not be the best idea, but we could consider putting it in some variable somewhere.